### DISCUSSION TITLE
Tasmota does not connect to WIFI

### DISCUSSION THREAD

**Discussion by atuxnull** [arendst/Tasmota#19082] [Q&A]
I got four sonoff basic and I flashed all of them with Tasmota(12.5). When I power on them and try to connect to the AP of the Tasmota under the name tasmota-xyz it does not provide IP to my laptop or phone. If I set manually the IP and try to connect to 192.168.4.1 I am able to get access to Tasmota's gui. Then I find my 2.4ghz wifi and set the password. Tasmota reboots and it is not able to connect to my WIFI. My wifi password is something abc!1xyz. 
It happens to all of the four devices that I have.  Any ideas on how to fix the issue, please?

**Comment by sfromis** [comment]
[From the docs](https://tasmota.github.io/docs/Commands/#password):
**Do not use special characters or white spaces in the password.**
